### PR TITLE
Replace deprecated APIs

### PR DESCRIPTION
**Pull request by offa**
Replacement of some deprecated APIs and migration of `try-fail-catch` based test cases to `assertThrows()`.

---------------------------

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelectorTest.java [hunk 1]
```diff
@@ -128,14 +128,12 @@ public static void m2(long x) {}
...
-                    new Object[]{new BooleanParameterValue("someBool", true), "x"}));
-        } catch (Exception e) {
-            assertTrue(e instanceof ClassCastException);
-            assertEquals("Cannot cast object 'x' with class 'java.lang.String' to class 'hudson.model.ParameterValue'",
-                    e.getMessage());
-        }
+        final Exception e = assertThrows(Exception.class,
+                () -> assertNull(GroovyCallSiteSelector.constructor(ParametersAction.class,
+                        new Object[]{new BooleanParameterValue("someBool", true), "x"})));
+        assertTrue(e instanceof ClassCastException);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelectorTest.java:134]:
```suggestion
        final ClassCastException e = assertThrows(ClassCastException.class,
                () -> assertNull(GroovyCallSiteSelector.constructor(ParametersAction.class,
                        new Object[]{new BooleanParameterValue("someBool", true), "x"})));
```
?

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java [hunk 2]
```diff
@@ -702,12 +695,8 @@ public static abstract class SpecialScript extends Script {
     }
 
     @Test public void ambiguousOverloads() {
-        try {
-            evaluate(new AnnotatedWhitelist(), Ambiguity.class.getName() + ".m(null)");
-            fail("Ambiguous overload is an error in Groovy 2");
-        } catch(GroovyRuntimeException e) {
-            // OK
-        }
+        assertThrows(GroovyRuntimeException.class,
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java:698]:
```suggestion
        assertThrows("Ambiguous overload is an error in Groovy 2", GroovyRuntimeException.class,
```

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java [hunk 3]
```diff
@@ -1434,14 +1406,10 @@ private void assertAnnotationBlocked(Class annotation, String script) {
...
-            shell.parse(script);
-            fail("Compilation should have failed");
-        } catch (Exception e) {
-            assertThat(e, instanceOf(MultipleCompilationErrorsException.class));
-            assertThat(e.getMessage(), anyOf(
-                    containsString("Annotation " + annotation.getName() + " cannot be used in the sandbox"),
-                    containsString("Annotation " + annotation.getSimpleName() + " cannot be used in the sandbox")));
-        }
+        final Exception e = assertThrows(Exception.class, () -> shell.parse(script));
+        assertThat(e, instanceOf(MultipleCompilationErrorsException.class));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java:1410]:
```suggestion
        final MultipleCompilationErrorsException e = assertThrows(MultipleCompilationErrorsException.class, () -> shell.parse(script));
```

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelectorTest.java (hunk 1 above)
Comment (offa) [reply src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelectorTest.java]: Thanks!

**Review by jglick — COMMENTED** [review]
merge conflict

**Comment by offa** [comment]
Resolved :+1:

**Review by jglick — APPROVED** [review]

**Pull request merged**